### PR TITLE
tests: add typeguard 4 compatibility

### PR DESCRIPTION
**Pull request by jelly**
The install_import_hook import changed in version 4.

### CODE REVIEW

**Comment by dcermak** [comment]
Thanks a lot for the pull request! I'd be more than happy to merge this, I'm just curious how you ran into this issue as `typeguard` is pinned to 2.13.3 in poetry.lock: https://github.com/dcermak/pytest_container/blob/d6f264bac300b20801bc8e833148c0c43369deb9/poetry.lock#L654

And out of interest, what are you using pytest_container for?

File: tests/conftest.py [hunk 1]
```diff
@@ -1,6 +1,9 @@
 # pylint: disable=missing-function-docstring,missing-module-docstring
 from typeguard import typechecked
-from typeguard.importhook import install_import_hook
+try:
+    from typeguard.importhook import install_import_hook
+except ImportError:
+    from typeguard import install_import_hook
```
Comment (dcermak) [inline tests/conftest.py:6]: This should make black happy again
```suggestion

try:
    from typeguard.importhook import install_import_hook
except ImportError:
    from typeguard import install_import_hook
```

**Comment by jelly** [comment]
> Thanks a lot for the pull request! I'd be more than happy to merge this, I'm just curious how you ran into this issue as `typeguard` is pinned to 2.13.3 in poetry.lock:
> 
> https://github.com/dcermak/pytest_container/blob/d6f264bac300b20801bc8e833148c0c43369deb9/poetry.lock#L654

Ah, I'm packaging it for Arch so build it with all the latest dependencies. Seems it might be an idea to bump typeguard?

> And out of interest, what are you using pytest_container for?

Nothing as of yet, but after seeing your devconf talk it seems interesting so first packaging it, we might use it for testing our container images.

**Comment by dcermak** [comment]
> > Thanks a lot for the pull request! I'd be more than happy to merge this, I'm just curious how you ran into this issue as `typeguard` is pinned to 2.13.3 in poetry.lock:
> > https://github.com/dcermak/pytest_container/blob/d6f264bac300b20801bc8e833148c0c43369deb9/poetry.lock#L654
> 
> Ah, I'm packaging it for Arch so build it with all the latest dependencies. Seems it might be an idea to bump typeguard?

Certainly! I do wonder why dependabot didn't take care of that though.

> 
> > And out of interest, what are you using pytest_container for?
> 
> Nothing as of yet, but after seeing your devconf talk it seems interesting so first packaging it, we might use it for testing our container images.

Ah, nice! Please let me know if there's any features that you're missing or any inconveniences that you encounter. Currently `pytest_container` fulfills all my needs, hence there's been very little development lately.

**Pull request merged**